pkg/kapp/clusterapply/change_set_view.go

```diff
@@ -48,3 +53,52 @@ func (v *ChangeSetView) Print(ui ui.UI) {
 func (v *ChangeSetView) Summary() string {
 	return v.changesView.Summary() // assumes Print was used before
 }
+
+func (v ChangeSetView) PrintCompleteYamlToBeApplied(ui ui.UI, conf ctlconf.Conf) error {
+	for _, view := range v.changeViews {
+		if view.ApplyOp() == ClusterChangeApplyOpNoop {
+			continue
+		}
+
+		opAndResDesc := ""
+		if view.ApplyOp() == ClusterChangeApplyOpDelete {
+			st, _ := view.ApplyStrategyOp()
+			if st == deleteStrategyPlainAnnValue {
+				opAndResDesc = color.RedString("# %s: %s", view.ApplyOp(), view.Resource().Description())
+			} else {
+				opAndResDesc = color.RedString("# %s %s: %s", st, view.ApplyOp(), view.Resource().Description())
+			}
+			ui.PrintBlock([]byte(opAndResDesc + "\n"))
+			continue
+		}
+		opAndResDesc = color.GreenString("# %s: %s", view.ApplyOp(), view.Resource().Description())
+
+		if view.ApplyOp() == ClusterChangeApplyOpExists {
+			ui.PrintBlock([]byte(opAndResDesc + color.GreenString(" => kapp will wait for this resource to be created\n")))
+			continue
+		}
+
+		resMgd := ctlres.NewResourceWithManagedFields(view.Resource(), false)
+		res, err := resMgd.Resource()
+		if err != nil {
+			return fmt.Errorf("Error: [%s]", err.Error())
+		}
+		if res.Kind() == "Secret" {
+			// masking secret
+			maskedRes := diff.NewMaskedResource(res, conf.DiffMaskRules())
+			res, err = maskedRes.Resource()
+			if err != nil {
+				return fmt.Errorf("Error: [%s]", err.Error())
+			}
+		}
+		by, err := res.AsYAMLBytes()
+		if err != nil {
+			return fmt.Errorf("Error: [%s]", err.Error())
+		}
+
+		ui.PrintBlock([]byte(opAndResDesc + "\n"))
+		ui.PrintBlock([]byte("---\n"))
+		ui.PrintBlock(by)
```

Comment:
There are so many `PrintBlock`s, maybe somehow we can consolidate them?

Reply:
updated